js/classes/entities/foreground/precipitation.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Creates falling rain or snow.
+ * 
+ * @author Caleb Krauter
+ */
+class Precipitation {
+
+    /**
+     * Based on the parameters a single raindrop or snowflake is added creating precipitation.
+     * 
+     * @param {*} dir direction that the drop should be falling.
+     * @param {*} pos initial position of the drop.
+     * @param {*} type rain or snow.
+     * @param {*} scale scale of the rain or snow will vary.
+     */
+    constructor(dir, pos, type, scale) {
+        this.dir = dir;
+        this.origin = pos;
+        // Setting this.pos to (0, 0) makes rain/snow fall naturally when it starts falling.
+        this.pos = new Vector(0, 0);
+        this.type = type;
+        this.velocity = new Vector(0, 0);
+        this.scale = scale;
+        this.imgIndex = 0;
+        // Precipitation.SPRITESHEET = Precipitation.SPRITESHEET_RAIN_DOWN;
+
+        if (type === "rain") {
```

Comment:
I would suggest making an enumeration that contains the possible values of this type variable. By entering just the string in these comparisons, it's a higher risk of a typo... human error. 

```js
// Near top of file
const weatherType = {
    RAIN: 'rain',
    SNOW: 'snow'
};

if (type === weatherType.RAIN) {
    console.log("It's raining.");
}
```

JavaScript doesn't natively support enumerations, but this is an example of what it would look like to use the enumeration design pattern. You just make a JavaScript map. This is useful in a lot of cases to reduce the risk of typos. It also makes it more obvious to others what possible options might exist for the type variable.

Reply:
Great idea, thanks for this!